shimmingtoolbox/masking/mask_mrs.py

```diff
@@ -0,0 +1,101 @@
+#!/usr/bin/python3
+# -*- coding: utf-8 -*-
+"""
+Creating MRS mask API
+"""
+import os
+import numpy as np
+import numpy.linalg as npl
+import nibabel as nib
+import logging
+import json
+import subprocess
+
+
+logging.basicConfig(level=logging.INFO)
+logger = logging.getLogger(__name__)
+
+
+def mask_mrs(fname_input, raw_data, X, Y, Z, V):
+    """
+    Create a mask to shim single voxel MRS
+
+    Args:
+        fname_input: Input path of the fieldmap to be shimmed (both nii.gz and json file should be in this path)
+        raw_data: Input path of the of the twix raw-data (supported extention .dat) [optional]
+        X: scanner's X position in mm [optional]
+        Y: scanner's Y position in mm [optional]
+        Z: scanner's Z position in mm [optional]
+        V: MRS voxel size in mm [optional]
+
+    Returns:
+        numpy.ndarray: Cubic mask with same dimensions as MRS voxel.
+    """
+
+    if fname_input is None:
+        raise FileNotFoundError(f"The file '{fname_input}' was not found. See: st_mask mask_mrs -h")
+
+    if X is not None and Y is not None and Z is not None and V is not None:
+        # If X,Y,Z,V arguments were provided, execute this block
+        MRS_voxel_thick = V
+        scanner_coordinate = np.array([X, Y, Z, 1])
+        logger.info(f"Scanner position: {scanner_coordinate}")
+
+    else:
+        # If X,Y,Z,V arguments were NOT provided, execute this block
+        if raw_data is None:
+            raise FileNotFoundError(f"The file '{raw_data}' was not found. If X,Y,Z and V are not directly given the raw-data must be given to read these info, See: st_mask mask_mrs -h")
+        else:
+            logger.info("Reading the twix raw-data")
+            subprocess.run(f"spec2nii twix {raw_data} -e image", shell=True )
+            name_nii = os.path.splitext(raw_data)[0]
+            header_twix= nib.load(name_nii + '.nii.gz').header
+            affine= nib.load(name_nii + '.nii.gz').affine
+            position_sag= header_twix['qoffset_x']
+            position_cor = header_twix['qoffset_y']
+            position_tra = header_twix['qoffset_z']
+            logger.info(f"twix header: {header_twix}")
+            logger.info(f"affine: {affine}")
+            scanner_coordinate= np.array([position_sag, position_cor, position_tra, 1])
+            logger.info(f"Scanner position: {scanner_coordinate}")
+
+
+    MRS_voxel_thick = header_twix['pixdim'][1]
```

Comment:
header_twix will not be assigned if the user specifies X, Y and Z. This would result in a crash.